source/transfer_list.rst

```diff
@@ -64,7 +66,7 @@ version field determines the contents of the handoff start header.
    * - checksum
      - 0x1
      - 0x4
-     - The checksum is used to ensure the data contained within the list is intact. The checksum is set to a value such that the sum over every byte in the {tl_base_pa, …, tl_base_pa + size -1} address range, modulo 256, is equal to 0.
+     - The checksum is used to ensure the data contained within the list is intact. The checksum is set to a value such that the sum over every byte in the {tl_base_pa + hdr_size, …, tl_base_pa + size - hdr_size -1} address range (i.e. the range of all TEs, including inter-TE padding, but excluding the TL header), modulo 256, is equal to 0.
```

Comment:
I think it should include the whole TL. If we exclude the TL header then there is no protection against corruption.

The checksum byte is clearly documented, so far as I can see. See the table where it describes the checksum.

Reply:
I'm of the opinion, the TL header should be part of the checksum computation.
The TL header is small, so the overhead will be minimal.
As Simon mention, this is also convenient since the consumer can detect when the header is corrupted or out of sync with the rest of the TL.

Reply:
"The checksum is set to a value such that the sum over every byte in the {tl_base_pa, …, tl_base_pa + size -1} address range, modulo 256, is equal to 0."

My understanding is that you set the checksum field to zero, calculate the checksum, then write the value into the checksum field. Perhaps we should be explicit about that?

Reply:
Okay, I'll change this to include the TL header in the checksum but clarify that the checksum field itself is assumed to be 0 for checksum calculation in the next version of this PR.

Reply:
Updated in latest push.